### PR TITLE
chore: use foundry.toml instead of Makefile

### PR DESCRIPTION
**Pull request by MathisGD**
Use `foundry.toml` instead of Makefile

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[profile.default]
```
Comment (MerlinEgalite) [inline foundry.toml:1]: I think we should set more parameters like solc version

Comment (MerlinEgalite) [reply foundry.toml:1]: Yes np, but I think it should be especially if we're doing gas snapshot to compare

Comment (MathisGD) [reply foundry.toml:1]: We are not doing this here. So should I do it ? If yes, with which version ?

File: foundry.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+src = "contracts"
+test = "test-foundry"
```
Comment (QGarchery) [inline foundry.toml:3]: Please insert a newline at the end of the file. There is a setting in VSCode to not have to worry about this: it is called "Files: Insert Final Newline"

**Review by QGarchery — APPROVED** [review]
Yes please no more makefile :)

File: Makefile [hunk 3]
```diff
@@ -1,19 +0,0 @@
...
-
-export FOUNDRY_TEST=test-foundry
-
-install:
-	@git submodule update --init --recursive
-	@yarn
-
-test:
-	@echo Run all tests
-	@forge test -vvv
```
Comment (QGarchery) [inline Makefile:11]: Both of those are advertised in the README, we can either keep them or change the doc

File: foundry.toml (hunk 1 above)
Comment (MathisGD) [reply foundry.toml:1]: Not sure to understand why, but ok why not.

File: Makefile (hunk 3 above)
Comment (MerlinEgalite) [reply Makefile:11]: It would be nice to also add in the doc how to assess gas consumption. There's nothing written written on this for now

Comment (MathisGD) [reply Makefile:11]: Do you feel that it is useful to tell people how to run tests when the setup is the one everyone uses ?

Comment (MathisGD) [reply Makefile:11]: Same question for you @MerlinEgalite. Everything is written in the foundry book already !

File: foundry.toml (hunk 1 above)
Comment (MerlinEgalite) [reply foundry.toml:1]: I thought you wanted to add gas snapshot on this repo. If not then you can forget about it

File: Makefile (hunk 3 above)
Comment (QGarchery) [reply Makefile:11]: We can the test section from the README, or keep part of the makefile. Either is ok for me

Comment (QGarchery) [reply Makefile:11]: (there is also the install section)

Comment (MerlinEgalite) [reply Makefile:11]: The REAMDE works for me with a mention to the test to check to compare gas is enough

**Review by QGarchery — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (MathisGD) [reply Makefile:11]: >The REAMDE works for me with a mention to the test to check to compare gas is enough

What do you mean ? There are test to compare the gas cost ? with what ?

Comment (MathisGD) [reply Makefile:11]: I updated the README

Comment (MerlinEgalite) [reply Makefile:11]: I think @QGarchery you were using a randomized test for gas comparison no?

Comment (QGarchery) [reply Makefile:11]: Yes `testFullHeapSort` is the best I feel. There is also `testHeapStructure` but a large part of the gas consumption is due to the checking so it's less reliable

Comment (MathisGD) [reply Makefile:11]: Ok I see. I dont think that it is worth having this in the README.

**Review by MerlinEgalite — APPROVED** [review]
LGTM

File: Makefile (hunk 3 above)
Comment (MerlinEgalite) [reply Makefile:11]: Ok as you want, it's not clear to me how an outside collaborator will know how to compare gas though

**Pull request merged**